From 57a08adf67ffb148cef502d795db15840a742eb9 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Fri, 26 Jul 2024 11:06:05 +0200
Subject: [zlib] Add missing extension for PMULL in ARM64 GCC

A regression was introduced in the macro TARGET_ARMV8_WITH_CRC for
GCC. In the past it was empty and it would relay on the passed
architecture to support the required extensions. After the change it
explicitely adds an attribute for functions calling PMULL that
states that they require the target armv8-a+crc.

But PMULL belongs to CRYPTO extension, that is not provided by default
in ARMv8-a. So build breaks. To fix it, add the "crypto" extension to
the macro setting the armv8-a target.

Bug: 40565911
Change-Id: I44f35813197801f0f8dfa032901864123db75468
---
 third_party/zlib/crc32_simd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/third_party/zlib/crc32_simd.c b/third_party/zlib/crc32_simd.c
index cbe9739578c94..1c60ae9bd3c11 100644
--- a/third_party/zlib/crc32_simd.c
+++ b/third_party/zlib/crc32_simd.c
@@ -398,7 +398,7 @@ uint32_t ZLIB_INTERNAL crc32_sse42_simd_(  /* SSE4.2+PCLMUL */
  */
 #include <arm_acle.h>
 #include <arm_neon.h>
-#define TARGET_ARMV8_WITH_CRC __attribute__((target("arch=armv8-a+crc")))
+#define TARGET_ARMV8_WITH_CRC __attribute__((target("arch=armv8-a+crc+crypto")))
 #else  // !defined(__GNUC__) && !defined(_aarch64__)
 #error ARM CRC32 SIMD extensions only supported for Clang and GCC
 #endif
-- 
2.43.0

